### PR TITLE
Prevent view annotation being shown erroneously after options update

### PR DESCRIPTION
**Pull request by evil159**
Updating options for an off-screen annotation with `visibility` set to `true` unhides the corresponding view which may result in it being displayed on top of the map. This PR addresses this issue by preventing updated view annotation to be un-hidden during the update. 
With this change view annotation are only shown as a result of view annotation position update event.

## Pull request checklist:
 - [x] Describe the changes in this PR, especially public API changes.
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add a changelog entry to to bottom of the relevant section (typically the `## main` heading near the top).

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: Sources/MapboxMaps/Annotations/ViewAnnotationManager.swift [hunk 1]
```diff
@@ -219,9 +219,12 @@ public final class ViewAnnotationManager {
         }
         let currentFeatureId = try? mapboxMap.options(forViewAnnotationWithId: id).associatedFeatureId
         try mapboxMap.updateViewAnnotation(withId: id, options: options)
-        let isHidden = !(options.visible ?? true)
-        expectedHiddenByView[view] = isHidden
-        viewsById[id]?.isHidden = isHidden
+
+        if let visible = options.visible, visible == false {
```
Comment (OdNairy) [inline Sources/MapboxMaps/Annotations/ViewAnnotationManager.swift:223]: It is the same, isn't it?
```suggestion
        if visible == false {
```

Comment (OdNairy) [reply Sources/MapboxMaps/Annotations/ViewAnnotationManager.swift:223]: And if it's not, can we add some tests to catch the corner case, please? This Optional<Bool> logic is really tricky 😅

Comment (evil159) [reply Sources/MapboxMaps/Annotations/ViewAnnotationManager.swift]: Same thing indeed, not sure why I went with more verbose syntax

**Review by OdNairy — APPROVED** [review]
